pkg/types/openstack/validation/machinepool.go

```diff
@@ -20,5 +23,26 @@ func ValidateMachinePool(p *openstack.MachinePool, fldPath *field.Path) field.Er
 		}
 	}
 
+	if len(p.AdditionalNetworkIDs) > 0 {
+		for i, uuid := range p.AdditionalNetworkIDs {
+			if !isValidUUID(uuid) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("additionalNetworkIDs"), p.AdditionalNetworkIDs, fmt.Sprintf("Invalid Network ID #%d", i)))
+			}
+		}
+	}
+
+	if len(p.AdditionalSecurityGroupIDs) > 0 {
+		for i, uuid := range p.AdditionalSecurityGroupIDs {
+			if !isValidUUID(uuid) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("additionalSecurityGroupIDs"), p.AdditionalSecurityGroupIDs, fmt.Sprintf("Invalid Security Group ID #%d", i)))
+			}
+		}
+	}
+
 	return allErrs
 }
+
+func isValidUUID(uuid string) bool {
+	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
```

Comment:
Can you link to the document that shows the  uuid_v4 needs to be this format.

Reply:
I think we already use https://godoc.org/github.com/google/uuid#Parse or something similar to generate cluster IDs...

so maybe we should do https://stackoverflow.com/a/46315070 or something like that instead of this regex match..

Reply:
we already https://github.com/openshift/installer/blob/b74e3919d38efad3361ac79fcb84958c823d329f/pkg/asset/installconfig/clusterid.go#L8 imported..

so you could use https://godoc.org/github.com/pborman/uuid#Parse ... and check for nil.. (https://github.com/pborman/uuid/blob/5b6091a6a160ee5ce12917b21ab96acec2a4fdc0/uuid.go#L60-L64)

Reply:
I decided to use "github.com/google/uuid" which is also in the list of requirements. It provides the same interface as "github.com/pborman/uuid".

Another check I added is to verify that parsed ids have version 4. Other versions are not supported by OpenStack.